### ISSUE TITLE
Improving issue handling process (was: Timestamp Problems with Comments / Activities and Trello import)

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#2724]
https://community.vanila.io/?t=f9c7dc70-4882-43ca-994e-153d0cca3af7

**Comment by erwin-k** [comment]
Wekan uses millisecond format on createdAt field, however when importing from trello it will use the returned value from new Date("trello's date") as seen here:

https://github.com/wekan/wekan/blob/afcca947e3f9ca14fb96a0548a3ceb90e9c16d15/models/trelloCreator.js#L54

which will return a string. 
I think this is what messes up sorting. 

What's weird is that Wekan uses millisecond format on createdAt but string format on other fields such as updatedAt or modifiedAt. So a simple .getTime() on the createdAt fields in trelloCreator.js will most likely fix this bug.

I can't get a dev environment setup, I tried but it does not work out for me so I can't test this but if somebody could try it out and make a pull request I would highly appreciate it!

**Comment by xet7** [comment]
@erwin-k 

For Dev environment, please try steps 3-5 on Linux (fastests) or Windows Subsystem for Linux (slow) :
https://github.com/wekan/wekan-maintainer/wiki/Developing-Wekan-for-Sandstorm

Thanks!

**Comment by erwin-k** [comment]
@xet7 

You could have told me this was already fixed in the latest version... I spend a lot of time getting Meteor dev environment running only to find out it's already been fixed.

**Comment by erwin-k** [comment]
@xet7 

You could have told me this bug was already fixed in the latest version... I spend a lot of time getting Meteor dev environment running only to find out this bug has already been fixed.

**Comment by xet7** [comment]
@erwin-k 

Great that you got Meteor dev environment running :) Then you can try fixing some other bug :) Thanks!

I did not know you did not have latest version.

**Comment by erwin-k** [comment]
@xet7 

Next time you should close issues as the bugs get fixed instead of wasting peoples time and telling them to "try fix some other bug :)". 

I won't be spending my time on this project anymore.

**Comment by justinr1234** [comment]
@erwin-k definitely understand the frustration. Running an open source project is pretty difficult to get right, particularly one as popular as this without any money funding it and with essentially only X doing a large majority of the work.

He misses stuff sometimes, particularly the very time consuming task of keeping up with Github issues. New issues are constantly submitted and different things are always in flight. It’s actually a bit of a full job just to keep up in the issues. But that’s on top of not getting paid for this, life responsibilities, actually fixing bugs himself, new features, fixing deployments, etc.

Knowing X I don’t think he meant to insult you, he was trying to look on the bright side of a bad situation. Any effort you put into the project is highly appreciated as we don’t have a lot of people contributing. We could definitely try to improve that through better developer experience and interactions. For instance in this situation, making your life easier by trying to come up with a better process to manage open issues and related bugs.

Do you know of any good automated ways to enforce a better process? I’m sure we could find one, but any help you know of would be great.

Again, completely understand it sucks to spend time on something that’s already fixed.

**Comment by xet7** [comment]
@justinr1234 

I already have access to GitHub Actions for automating. How should process be improved?

Maybe I should read [Jobo Bacon's book](https://twit.tv/shows/twit-bits/episodes/5913) to understand more about this. I think currently it's too easy for someone to misunderstand how things work.

**Comment by xet7** [comment]
Some current problems:
- I have not updated FAQ for some time, so I end up answering same questions again.
- Documentation is not organized and up-to-date enough, so there is questions where something is.

**Comment by justinr1234** [comment]
Some idea would be this: https://github.com/probot/stale

https://github.com/probot/duplicate-issues

**Comment by xet7** [comment]
Stale issues: No. There are very important old issues that should not be closed, and they don't receive comments usually.

Duplicate issues: I'm not sure how that works, and how safe it is to run. How can I make sure it does not mess up issues.

I need to think carefully before implementing any of those.

**Comment by justinr1234** [comment]
@xet7 there's likely a config that will say "should this be closed" and then allow us to intervene

**Comment by xet7** [comment]
@justinr1234 

I will sometime look is there duplicate issue detection etc at GitHub issues, and if not, look at probot. And test them.

**Comment by erwin-k** [comment]
![image](https://cdn.discordapp.com/attachments/549068199044317185/629091062975037451/unknown.png)

This merge was done for this exact bug.
All that had to be done is close this issue, but instead for some reason this issue was created after this bug was already fixed.

Maybe not using Vanila.io for bug reports is an idea, as this issue was opened a month after it was posted on Vanila.io and 16 days after it had already been fixed.
In total it's three and a half weeks where people like me could have been googling this and even after I responded to this issue it is still not realized that this bug has long been fixed. Instead I get to install linux and spend time on getting Meteor running just to find out I've been wasting my time.

This will solve your problem:
1) Bug appears
2) Open issue if no issue yet
3) Reference this issue in any future duplicate issues
4) Bug get's fixed
5) Close all issues related to this bug
6) Next task
